client/tailscale/localclient.go

```diff
@@ -41,6 +41,13 @@ import (
 	"tailscale.com/types/tkatype"
 )
 
+// DialFunc is any function that dials the given address.
+type DialFunc = func(ctx context.Context, network, addr string) (net.Conn, error)
+
+// h2cTransport is non-nil on platforms where we want to add H2C
+// ("cleartext" HTTP/2) support to the LocalAPI.
+var h2cTransport func(DialFunc) http.RoundTripper
```

Comment:
Instead of `h2cTransport` being `nil`, could it just be an empty func?

Reply:
It can't be an empty function because, unlike the server-side addH2C, which configures a given `http.Server` to support h2c, this function _returns_ an h2c transport to be used by the client.

I updated it to _return_ `nil`, rather than _being_ `nil`, when h2c is not supported on the current platform, but am not sure if that's any better.